src/web/app/components/sortable-table/sortable-table.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { TableComparatorService } from '../../../services/table-comparator.service';
+import { SortBy, SortOrder } from '../../../types/sort-properties';
+
+/**
+ * Column data for sortable table
+ */
+export interface ColumnData {
+  header: string;
+  headerToolTip?: string;
+  sortBy?: SortBy; // Don't provide this field if you don't want this column to be sortable
+}
+
+/**
+ * Displays a sortable table, sorting by clicking on the header
+ * Optional sortBy option provided for each column
+ * Columns and rows provided must be aligned
+ */
+@Component({
+  selector: 'tm-sortable-table',
+  templateUrl: './sortable-table.component.html',
+  styleUrls: ['./sortable-table.component.scss'],
+})
+export class SortableTableComponent implements OnInit {
+
+  @Input()
+  columns: ColumnData[] = [];
+
+  @Input()
+  rows: any[][] = [];
+
+  columnToSortBy: string = '';
+  sortOrder: SortOrder = SortOrder.ASC;
+  tableRows: any[][] = [];
+
+  // enum
+  SortOrder: typeof SortOrder = SortOrder;
```

Comment:
To be consistent with other files, we put enum declaration at the top of class.

Reply:
Actually there's some inconsistencies between files for this. But majority put it at the top, so I'll make this change as well.